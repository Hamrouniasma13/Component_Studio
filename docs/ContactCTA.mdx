---
name: Contact CTA
route: /contact-cta
menu: 🌐 Organisms
---

## Type: `Organism`

import { Playground } from 'docz'
import { ContactCTA, Button } from '../src'

# ContactCTA

<Playground>
  <ContactCTA
    title="Title"
    content="Pariatur et exercitation laborum cillum minim laborum laborum veniam eu dolor. Irure eiusmod in officia tempor tempor exercitation incididunt exercitation fugiat et aliquip ex minim. Ex elit dolor incididunt consectetur do velit ipsum elit. Incididunt voluptate pariatur deserunt mollit ipsum."
    image="https://prismic-io.s3.amazonaws.com/mini-site%2F2b35a26c-7599-47b4-b962-3eb57379a147_logo.png"
    action={<Button href="/">Call to action</Button>}
  />
</Playground>
