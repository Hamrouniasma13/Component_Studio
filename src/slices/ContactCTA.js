/** @jsx jsx */
import { jsx, ThemeContext } from '@emotion/core'
import React from 'react'
import PropTypes from 'prop-types'

import Slice from './Slice'
import Text from '../components/Text'

export default function ContactCTA({ title, content, image, action }) {
  const theme = React.useContext(ThemeContext)

  const styles = {
    box: {
      minHeight: 140,
      boxShadow: '0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)',
      padding: 30,
      backgroundColor: '#fff',
      borderRadius: 3,
      [theme.media.laptop]: {
        display: 'flex',
        alignItems: 'center'
      }
    },
    image: {
      display: 'none',
      maxWidth: 80,
      maxHeight: 80,
      [theme.media.laptop]: {
        display: 'inline-block',
        marginRight: 40
      }
    },
    content: {
      [theme.media.laptop]: {
        flex: 1,
        alignSelf: 'flex-start'
      }
    },
    action: {
      marginTop: 15,
      [theme.media.laptop]: {
        marginTop: 0,
        marginLeft: 40
      }
    }
  }

  return (
    <Slice title={title}>
      <div css={styles.box}>
        {image && <img css={styles.image} src={image} alt="contact logo" />}
        <Text tag="div" css={styles.content}>
          {content}
        </Text>
        {action && <div css={styles.action}>{action}</div>}
      </div>
    </Slice>
  )
}

ContactCTA.propTypes = {
  title: Slice.propTypes.title,
  content: PropTypes.any,
  image: PropTypes.string,
  action: PropTypes.element
}
