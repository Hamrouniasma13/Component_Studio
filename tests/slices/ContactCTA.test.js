import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

import ContactCTA from '../../slices/ContactCTA';

describe('Component: <NavigationBanner />', () => {
  it('render snapshot', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <ContactCTA
          content="Et velit officia nulla nostrud enim Lorem dolor et laboris ut consectetur."
          link={{
            url: '/',
            label: 'accÃ¨s au service'
          }}
          image={{
            url:
              'https://prismic-io.s3.amazonaws.com/mini-site%2F2b35a26c-7599-47b4-b962-3eb57379a147_logo.png',
            alt: 'My logo'
          }}
        />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
