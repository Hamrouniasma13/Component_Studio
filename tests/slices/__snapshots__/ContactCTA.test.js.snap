// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <NavigationBanner /> render snapshot 1`] = `
.emotion-6 {
  background-color: white;
  text-align: center;
  padding-top: 50px;
  padding-bottom: 50px;
}

.emotion-5 {
  margin: 0px auto;
  max-width: 100%;
  padding-left: 15px;
  padding-right: 15px;
}

@media only screen and (min-width :768px) {
  .emotion-5 {
    width: 720px;
  }
}

@media only screen and (min-width :992px) {
  .emotion-5 {
    width: 960px;
  }
}

@media only screen and (min-width :1200px) {
  .emotion-5 {
    width: 1140px;
  }
}

.emotion-4 {
  min-height: 140px;
  padding-left: 30px;
  padding-right: 30px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  padding-top: 30px;
  padding-bottom: 30px;
  background-color: white;
  border-radius: 4px;
}

@media only screen and (min-width :768px) {
  .emotion-4 {
    padding-left: 30px;
    padding-right: 30px;
    position: relative;
  }
}

.emotion-0 {
  margin-bottom: 15px;
  max-width: 80px;
}

@media only screen and (min-width :992px) {
  .emotion-0 {
    float: left;
  }
}

.emotion-1 {
  font-size: 16px;
  line-height: 1.5;
  font-family: Source Sans Pro,Arial,Helvetica,sans-serif;
  color: black;
  margin-bottom: 0px;
}

.emotion-1 p > a {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #494df4;
  font-weight: bold;
}

@media only screen and (min-width :768px) {
  .emotion-1 {
    font-size: 18px;
    line-height: 1.67;
  }
}

@media only screen and (min-width :992px) {
  .emotion-1 {
    text-align: left;
    margin-left: 120px;
    max-width: 540px;
  }
}

.emotion-3 {
  margin-top: 30px;
}

@media only screen and (min-width :992px) {
  .emotion-3 {
    float: right;
    display: block;
    top: 50%;
    position: absolute;
    right: 30px;
    bottom: 30px;
    margin-top: -20px;
  }
}

.emotion-2 {
  cursor: pointer;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-weight: bold;
  background-color: #494df4;
  color: #ffffff;
  font-family: Source Sans Pro,Arial,Helvetica,sans-serif;
  font-size: 15px;
  border-radius: 4px;
  -webkit-letter-spacing: 1.1px;
  -moz-letter-spacing: 1.1px;
  -ms-letter-spacing: 1.1px;
  letter-spacing: 1.1px;
  padding: 0 20px;
  position: relative;
  z-index: 0;
  overflow: hidden;
  line-height: 40px;
  text-align: center;
  max-width: 700px;
  vertical-align: middle;
  display: inline-block;
  border: none;
}

.emotion-2:after {
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  background-color: #000014;
  content: "";
  height: 530px;
  left: 50%;
  opacity: 0;
  position: absolute;
  top: 50%;
  -webkit-transform: translate3d(-50%,-50%,0) rotate(45deg);
  -ms-transform: translate3d(-50%,-50%,0) rotate(45deg);
  transform: translate3d(-50%,-50%,0) rotate(45deg);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  width: 0;
  z-index: -1;
}

.emotion-2:hover:after {
  opacity: 1;
  -webkit-transform: translate3d(-50%,-50%,0) rotate(45deg);
  -ms-transform: translate3d(-50%,-50%,0) rotate(45deg);
  transform: translate3d(-50%,-50%,0) rotate(45deg);
  width: 100%;
}

<div
  className="emotion-6"
>
  <div
    className="emotion-5"
  >
    <div
      className="emotion-4"
    >
      <img
        alt="My logo"
        className="emotion-0"
        src="https://prismic-io.s3.amazonaws.com/mini-site%2F2b35a26c-7599-47b4-b962-3eb57379a147_logo.png"
      />
      <p
        className="emotion-1"
      >
        Et velit officia nulla nostrud enim Lorem dolor et laboris ut consectetur.
      </p>
      <div
        className="emotion-3"
      >
        <a
          aria-current="page"
          className="Button  emotion-2 active"
          color="primary"
          href="/"
          onClick={[Function]}
          size="normal"
          style={Object {}}
        >
          acc√®s au service
        </a>
      </div>
    </div>
  </div>
</div>
`;
